### PR TITLE
Minor beautifications salvaged from #5390

### PR DESCRIPTION
**Pull request by mih**
I consider those useful, even when no longer going forward with #5390

### CODE REVIEW

File: datalad/core/local/create.py [hunk 1]
```diff
@@ -516,3 +457,115 @@ def __call__(
...
+        'type': 'file',
+        'state': 'untracked'}
+    return tbrepo
+
+
+def _setup_annex_repo(path, add_to_git, initopts=None, fake_dates=False,
+                      description=None):
+    """Create and configure a repository at `path`
+
+    This includes a default setup which files to never annex.
```
Comment (kyleam) [inline datalad/core/local/create.py:501]: s/which/of &/ ?

Or perhaps "This includes a default setup of `annex.largefiles`."

File: datalad/core/local/create.py [hunk 2]
```diff
@@ -516,3 +457,115 @@ def __call__(
...
+
+
+def _setup_git_repo(path, add_to_git, initopts=None, fake_dates=False):
+    """Create and configure a repository at `path`
+
+    Parameters
+    ----------
+    path: str or Path
+      Path of the repository
+    add_to_git: dict
```
Comment (kyleam) [inline datalad/core/local/create.py:469]: `_setup_git_repo` and `_setup_annex_repo` don't care about what information is already in `add_to_git`, so I'd find it clearer if the side channel was avoided by each function dropping the parameter and returning a freshly created dictionary alongside the repo instance.

**Review by kyleam — APPROVED** [review]
Thanks, I think this makes `create` much more digestible.

Left a comment about `add_to_git`, but I have no objections if things are left as is.

File: datalad/core/local/create.py (hunk 1 above)
Comment (mih) [reply datalad/core/local/create.py:501]: Yes, true, done (the full replacement).

File: datalad/core/local/create.py (hunk 2 above)
Comment (mih) [reply datalad/core/local/create.py:469]: Done!

**Pull request merged**